src/main/java/uk/gov/justice/hmpps/datacompliance/services/referral/ReferralResolutionService.java

```diff
@@ -64,11 +62,12 @@ public void processReferral(final OffenderDeletionReferral referral,
 
     public void processUpdatedRetentionCheck(final RetentionCheck retentionCheck) {
 
-        final var referralResolution = retentionCheck.getReferralResolution();
+        final var referralResolution =
+                referralResolutionRepository.findById(retentionCheck.getReferralResolution().getResolutionId())
+                        .orElseThrow();
```

Comment:
By finding the 'resolution' entity using the id, it is now within the persistence context and we can lock the entity.